components/System/DarkModeToggle.vue

```diff
@@ -35,18 +35,18 @@ export default Vue.extend({
     }
   },
   computed: {
-    ...mapGetters('displayPreferences', ['getBooleanCustomPref']),
+    ...mapGetters('displayPreferences', ['getDarkMode']),
```

Comment:
Why a getter for this? Imo it would be better to do the conversion to boolean before committing and map the state everywhere

Reply:
I agree, I think it'd be better having the state be the real typed value and mapping the string/boolean at fetch/update time. I can do this

Reply:
Actually, it's pretty hard to not hardcode the boolean code for those values

Reply:
@ThibaultNocchi I don't understand what you mean ^^"

Reply:
Yeah understandable haha. If the store state contains a boolean, it means we have to parse the dark mode value at fetch and update times. The only solution for this is to hardcode the cases for each boolean value we'll have, I can't make a list of keys such as `['darkMode', ...]` and iterate over the state keys (TS gives me an error that we can't use string keys on a DisplayPreferencesState). This means code like this doesn't pass the linter:

```javascript
Object.keys(state).forEach(key=>{
  const whatever = state[key] // state[key] doesn't pass
})
```

I prefer the current PR way than hardcode the mapping of each boolean value in the fetch and update commits. :)

Reply:
Hmm, I'll take a look at it, maybe I can find a solution.

Reply:
It could be great to get this in soon as it doesn't work currently on SSR :)